pandas/tests/arrays/categorical/test_missing.py

```diff
@@ -100,6 +100,13 @@ def test_fillna_iterable_category(self, named):
 
         tm.assert_categorical_equal(result, expected)
 
+        # Case where the Point is not among our categories; we want ValueError,
+        #  not NotImplementedError GH#41914
+        cat = Categorical(np.array([Point(1, 0), Point(0, 1), None], dtype=object))
+        msg = "Cannot setitem on a Categorical with a new category"
```

Comment:
is it possible to customize the error messages to the caller method (eg fill a here) as we have a generic validate (good)

can be a follow on

Reply:
yah thatd be nice.  we did something like that for the datetimelikes